paths_cli/tests/compiling/test_volumes.py

```diff
@@ -0,0 +1,118 @@
+            'external': 'foo'
+        }
+
+    def create_inputs(self, inline, periodic):
+        yml = "\n".join(["type: cv-volume", "cv: {func}",
+                         "lambda_min: 0", "lambda_max: 1"])
+
+    def set_periodic(self, periodic):
+        if periodic == 'periodic':
+            self.named_objs_dict['foo']['period_max'] = 'np.pi'
+            self.named_objs_dict['foo']['period_min'] = '-np.pi'
+
+    @pytest.mark.parametrize('inline', ['external', 'external'])
+    @pytest.mark.parametrize('periodic', ['periodic', 'nonperiodic'])
+    def test_build_cv_volume(self, inline, periodic):
+        self.set_periodic(periodic)
+        yml = self.yml.format(func=self.func[inline])
+        dct = yaml.load(yml, Loader=yaml.FullLoader)
+        if inline =='external':
+            patch_loc = 'paths_cli.compiling.root_compiler._COMPILERS'
+            compilers = {
+                'cv': mock_compiler('cv', named_objs={'foo': self.mock_cv})
+            }
+            with mock.patch.dict(patch_loc, compilers):
+                vol = build_cv_volume(dct)
+        elif inline == 'internal':
+            vol = build_cv_volume(dct)
+        assert vol.collectivevariable(1) == 0.5
+        expected_class = {
+            'nonperiodic': paths.CVDefinedVolume,
+            'periodic': paths.PeriodicCVDefinedVolume
+        }[periodic]
+        assert isinstance(vol, expected_class)
+
+
+class TestBuildCombinationVolume:
+    def setup(self):
+        from  openpathsampling.experimental.storage.collective_variables \
+                import CollectiveVariable
+        self.cv = CollectiveVariable(lambda s: s.xyz[0][0]).named('foo')
+
+    def _vol_and_yaml(self, lambda_min, lambda_max, name):
+        yml = ['- type: cv-volume', '  cv: foo',
+               f"  lambda_min: {lambda_min}",
+               f"  lambda_max: {lambda_max}"]
+        vol = paths.CVDefinedVolume(self.cv, lambda_min,
+                                    lambda_max).named(name)
+        description = {'name': name,
+                       'type': 'cv-volume',
+                       'cv': 'foo',
+                       'lambda_min': lambda_min,
+                       'lambda_max': lambda_max}
+        return vol, yml, description
+
+    @pytest.mark.parametrize('combo', ['union', 'intersection'])
+    @pytest.mark.parametrize('inline', [True, False])
+    def test_build_combo_volume(self, combo, inline):
+        vol_A, yaml_A, desc_A = self._vol_and_yaml(0.0, 0.55, "A")
+        vol_B, yaml_B, desc_B = self._vol_and_yaml(0.45, 1.0, "B")
+        if inline:
+            named_volumes_dict = {}
+            descriptions = {}
+            subvol_yaml = ['  ' + line for line in yaml_A + yaml_B]
+        else:
+            named_volumes_dict = {v.name: v for v in [vol_A, vol_B]}
+            descriptions = {"A": desc_A, "B": desc_B}
+            subvol_yaml = ['  - A', '  - B']
+
+        yml = "\n".join(["type: {combo}", "name: bar", "subvolumes:"]
```

Comment:
missing `f` to make an f-string? (How would this pass???)
```suggestion
        yml = "\n".join([f"type: {combo}", "name: bar", "subvolumes:"]
```

Reply:
It passed because this didn't actually test that the combo type was being parsed (and dispatched to the right function) correctly -- an integration test with more scope than needed here. This is just the unit test that parsing a union/intersection actually creates the union/intersection correctly.

(But I had to spend some time with the code to figure that out!) Definitely accepting this commit suggestion.